core/src/main/scala/org/apache/spark/Dependency.scala

```diff
@@ -206,6 +206,21 @@ class ShuffleDependency[K: ClassTag, V: ClassTag, C: ClassTag](
     finalizeTask = Option(task)
   }
 
+  // Set the threshold to 1 billion which represents approximately 1GB of memory
+  // allocated to map output statuses.
+  // A large number of shuffle blocks may crash the driver with an OOM error.
+  private val SHUFFLE_BLOCK_NUMBER_WARNING_THRESHOLD: Long = 1000000000L
+  private val numberOfShuffleBlocks = numPartitions.toLong * partitioner.numPartitions.toLong
+
+  if (numberOfShuffleBlocks > SHUFFLE_BLOCK_NUMBER_WARNING_THRESHOLD) {
+    logWarning(
+      s"The number of shuffle blocks (${numberOfShuffleBlocks}) for ${_rdd} " +
```

Comment:
Maybe we could also log the shuffle ID here as well? e.g. 

```
"The number of shuffle blocks (...) for shuffleId $shuffleId for ${_rdd} with ${numPartition} partitions ...."
```

or something along those lines?